crates/ruff_linter/resources/test/fixtures/flake8_pyi/PYI049.pyi

```diff
@@ -35,3 +35,12 @@ _UnusedTypedDict3 = TypedDict("_UnusedTypedDict3", {"foo": int})
 _UsedTypedDict3 = TypedDict("_UsedTypedDict3", {"bar": bytes})
 
 def uses_UsedTypedDict3(arg: _UsedTypedDict3) -> None: ...
+
+
+# Error
+class _CustomClass3:
+    class _UnusedTypeDict4(TypedDict):
+        pass
+
+    def f(self) -> None:
+        print(_CustomClass3._UnusedTypeDict4())
```

Comment:
Nit: maybe get rid of the `print()` call in the `.pyi` fixture here; I'm not sure it adds anything

```suggestion
# Error
# (in .pyi files, we flag unused definitions in class scopes
# as well as the global scope, unlike in .py files)
class _CustomClass3:
    class _UnusedTypeDict4(TypedDict):
        pass
```